### PR TITLE
JS: Use different ES target version for node.js in forum sample tests

### PR DESCRIPTION
**Pull request by letmaik**
Addresses https://github.com/microsoft/CCF/issues/1796#issuecomment-715897771.

### CODE REVIEW

**Review by eddyashton — APPROVED** [review]

File: samples/apps/forum/tsconfig.json [hunk 1]
```diff
@@ -11,6 +11,12 @@
         "preserveConstEnums": true,
         "sourceMap": false
     },
+    "ts-node": {
+        "compilerOptions": {
+            "lib": ["ES2015"],
```
Comment (achamayou) [inline samples/apps/forum/tsconfig.json:16]: Why ES2015?

Comment (letmaik) [reply samples/apps/forum/tsconfig.json:16]: Because everything is green for ES2015: https://node.green/

**Comment by letmaik** [comment]
I just tested this again and it doesn't work anymore... let me check this. Spooky.

**Comment by letmaik** [comment]
> 
> 
> I just tested this again and it doesn't work anymore... let me check this. Spooky.

Alright, it's working now. I replaced `ts-mocha` by using `mocha` directly with the `-r ts-node/register` option.

File: samples/apps/forum/tsconfig.json (hunk 1 above)
Comment (achamayou) [reply samples/apps/forum/tsconfig.json:16]: Oh I see, I was confused and though this was to do with QuickJS, nevermind.

**Comment by achamayou** [comment]
@letmaik halloween.js

**Pull request merged**